From 36459e3f00c1e9997a224e30ff7b191f06c6b3f8 Mon Sep 17 00:00:00 2001
From: Evan Wang <xswang@marvell.com>
Date: Wed, 30 Aug 2017 19:08:46 +0800
Subject: [PATCH 1977/2241] fix: dts: cp110: skip comphy power off for PCIe
 lanes

PCIe comphy driver introduced an issue with detection of
SATA disks over PCIe.

This patch is a WA for this issue, which skips PCIe COMPHY
power off (triggered in COMPHY probe stage). With this WA,
and without associating PCIe to their COMPHY lanes, proper
functionality of PCIe COMPHY lanes depends on U-Boot
initialization, and these will be removed once SATA
detection issues will be resolved.

Change-Id: I410a00448f6e581b541fd93f206510a8c0fc3d74
Signed-off-by: Evan Wang <xswang@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/43549
Tested-by: iSoC Platform CI <ykjenk@marvell.com>
Reviewed-by: Omri Itach <omrii@marvell.com>
---
 arch/arm64/boot/dts/marvell/armada-cp110-1.dtsi | 1 +
 arch/arm64/boot/dts/marvell/armada-cp110.dtsi   | 1 +
 2 files changed, 2 insertions(+)

diff --git a/arch/arm64/boot/dts/marvell/armada-cp110-1.dtsi b/arch/arm64/boot/dts/marvell/armada-cp110-1.dtsi
index 972dddd..88fc93f 100644
--- a/arch/arm64/boot/dts/marvell/armada-cp110-1.dtsi
+++ b/arch/arm64/boot/dts/marvell/armada-cp110-1.dtsi
@@ -393,6 +393,7 @@ comphy1: comphy {
 	dfx-reg = <0x400280 4>;
 	#phy-cells = <2>;
 	sar-data = <&cps_sar>;
+	skip_pcie_power_off;
 	status = "disabled";
 };
 
diff --git a/arch/arm64/boot/dts/marvell/armada-cp110.dtsi b/arch/arm64/boot/dts/marvell/armada-cp110.dtsi
index 67dd74e..8daa36f 100644
--- a/arch/arm64/boot/dts/marvell/armada-cp110.dtsi
+++ b/arch/arm64/boot/dts/marvell/armada-cp110.dtsi
@@ -503,6 +503,7 @@ comphy0: comphy {
 	dfx-reg = <0x400280 4>;
 	#phy-cells = <2>;
 	sar-data = <&cpm_sar>;
+	skip_pcie_power_off;
 	status = "disabled";
 };
 
-- 
2.7.4

